### PR TITLE
insights: enable aggregations on single docker deployments

### LINKED ISSUE #40368: insights: enable search aggregations on single docker deployments
Currently all code insights graphql resolvers are disabled when they are run in single docker deployment mode.  

Search aggregations should be enabled in this mode.  Update the initialization of insights resolvers to enable the running of search aggregation query even if the deployment is single docker.

/cc @joelkw @felixfbecker @vovakulikov @unclejustin

### PR DESCRIPTION
**Pull request by leonore**
closes #40368 

## Test plan

covered in detail in the [spreadsheet](https://docs.google.com/spreadsheets/d/11aQqWEVe_S11Ff_W9ntjjLwaV9XsfkmhuJGZ2z8wqAQ/edit#gid=1699079299)

```sh
# build the docker image
sg ci build docker-image-candidates-notest

# fetch the image url from the logs/environment
IMAGE="url"
docker run --publish 7080:7080 --rm $IMAGE
docker run -e ALLOW_SINGLE_DOCKER_CODE_INSIGHTS=true --publish 7080:7080 --rm $IMAGE
```

to create an enterprise license, navigate to https://sourcegraph.com/site-admin/dotcom/product/subscriptions

### CODE REVIEW

File: enterprise/cmd/frontend/internal/batches/resolvers/main_test.go [hunk 1]
```diff
@@ -118,6 +119,10 @@ func parseJSONTime(t testing.TB, ts string) time.Time {
 	return timestamp
 }
 
+func newSchema(db database.DB, r graphqlbackend.BatchChangesResolver) (*graphql.Schema, error) {
```
Comment (leonore) [inline enterprise/cmd/frontend/internal/batches/resolvers/main_test.go:122]: if a batcher is passing by: I've moved this function out of `resolver_test` so it's reused across tests where relevant. this should make any future changes to our graphql schema a lot easier for changing your tests!

**Comment by chwarwick** [comment]
I think this approach is good.

>  I'm using the same insights schemas for the two resolvers

By that are you referring to leaving the aggregations schema and structs in the `insights.graphql` and `insights.go` files?  I think it would be nice to move them out as a reminder to us that they are different and have different availability but certainly not a blocking concern.

**Comment by unknwon** [comment]
Congrats on merging 😁  qq: does this mean we now have code insights in single-docker deployment?

**Comment by leonore** [comment]
> Congrats on merging 😁 qq: does this mean we now have code insights in single-docker deployment?

code insights have always been "allowed" on single docker deployments but discouraged. you can enable them with the `ALLOW_SINGLE_DOCKER_CODE_INSIGHTS` env variable

aggregations are allowed regardless of that env variable

**Pull request merged**